File: src/main/java/com/zuminX/annotations/swagger/Authorization.java
package com.zuminX.annotations.swagger;

import com.zuminX.annotations.AnnotationAttr;
import com.zuminX.annotations.AnnotationStr;
import com.zuminX.names.ClassName;
import com.zuminX.names.SwaggerAnnotation;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * 对应io.swagger.annotations.Authorization
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Authorization extends AnnotationStr {

  @AnnotationAttr(show = true)
  private String value;

  @AnnotationAttr(show = true)
  private List<AuthorizationScope> scopes;

  @Override
  public ClassName getClassName() {
    return SwaggerAnnotation.AUTHORIZATION;
  }
}

File: src/main/java/com/zuminX/annotations/swagger/Authorization.kt
package com.zuminX.annotations.swagger

import com.zuminX.annotations.AnnotationAttr
import com.zuminX.annotations.AnnotationStr
import com.zuminX.names.ClassName
import com.zuminX.names.annotation.AUTHORIZATION

/**
 * 对应io.swagger.annotations.Authorization
 */
class Authorization : AnnotationStr() {

  @field:AnnotationAttr(show = true)
  var value: String? = null

  @field:AnnotationAttr(show = true)
  var scopes: List<AuthorizationScope>? = null
  
  override fun getClassName(): ClassName = AUTHORIZATION
}